MetaMask/metamask-extension
# Part of #18714 and #17670:set-approval-for-all-warnings

## Explanation
Replaces deprecated Typography and Button components in `ui\components\app\set-approval-for-all-warning\set-approval-for-all-warning.js` as well as fixes strong tag issue in Text component

Part of

- https://github.com/MetaMask/metamask-extension/issues/18714
- https://github.com/MetaMask/metamask-extension/issues/17670
- https://github.com/MetaMask/metamask-extension/issues/18896

## Screenshots/Screencaps

### Before

<img width="1440" alt="Screenshot 2023-05-15 at 9 21 41 AM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/51c81785-7ceb-4177-9b3a-760faa42edcb">

### After

<img width="1440" alt="Screenshot 2023-05-15 at 9 20 52 AM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/98fef390-756d-4536-868e-07862b22e66c">

## Manual Testing Steps

- Pull this branch
- Run `yarn storybook`
- Check the `SetApproveForAllWarning` story

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
